src/main/java/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,53 @@
+package bookstore.repository.impl;
+
+import bookstore.entity.Book;
+import bookstore.exception.DataProcessingException;
+import bookstore.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
As of Spring Framework 4.3, an @Autowired annotation on such a constructor is no longer necessary if the target bean defines only one constructor to begin with. See [note here ](https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html)Let’s check it ;)